includes/admin/reports/class-wc-admin-report-v2.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * V2 report functionality
+ *
+ * @package WooCommerce\Admin\Reports
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+if ( ! class_exists( 'WC_Admin_Report' ) ) {
+	require_once dirname( __FILE__ ) . '/class-wc-admin-report.php';
+}
+
+/**
+ * Class WC_Admin_Report_V2
+ * Makes use of the WC Admin plugin endpoints if it is not disabled.
+ */
+class WC_Admin_Report_V2 extends WC_Admin_Report {
```

Comment:
I can see why you've taken this approach. After some thought I wonder, to avoid migration confusion or complicated missions to deprecate classes in future and general mess of naming classes things like `-v2` what would be the downside of having this functionality be a private function in the existing class and call it if wc-admin is active from within the original method?

The big benefit I see there is that we avoid having to mark classes deprecated, migrate existing calls to this new class, or suffix classes with version numbers. 

What do you think?

Reply:
Actually this suggestion was a bit naive since the existing calls to this method will want to have the original behavior in place despite whether wc admin is active.

Reply:
Technically only for the `sales_sparkline` function, which could be moved to a private function similar to what you mentioned above.
But, given the current logic will slowly get phased out by making use of the new WooAdmin logic, I figured it might be nice to have a central place (kinda like a funnel) so that when we do pull the switch it won't be too much of a hassle.
I could also create a separate class that handles all this instead, but I had trouble naming it, that's why I went with the V2 part.

Reply:
@samueljseay I end up moving the logic back to some private functions, and wrote some tests for the entire `status_widget`, see the changes here: https://github.com/woocommerce/woocommerce/pull/29149/commits/966aca59ad9523dc98d218287f2be544877eedc1 (although easier to look at the main diff).